File: seed/rust-sdk/trace/src/types/create_problem_request.rs
use crate::problem_description::ProblemDescription;
use crate::language::Language;
use crate::problem_files::ProblemFiles;
use crate::variable_type_and_name::VariableTypeAndName;
use crate::variable_type::VariableType;
use crate::test_case_with_expected_result::TestCaseWithExpectedResult;
use std::collections::HashMap;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct CreateProblemRequest {
    #[serde(rename = "problemName")]
    pub problem_name: String,
    #[serde(rename = "problemDescription")]
    pub problem_description: ProblemDescription,
    pub files: HashMap<Language, ProblemFiles>,
    #[serde(rename = "inputParams")]
    pub input_params: Vec<VariableTypeAndName>,
    #[serde(rename = "outputType")]
    pub output_type: VariableType,
    pub testcases: Vec<TestCaseWithExpectedResult>,
    #[serde(rename = "methodName")]
    pub method_name: String,
}

File: seed/ruby-sdk-v2/trace/lib/seed/types/create_problem_request.rb
# frozen_string_literal: true

module Seed
  module Types
    class CreateProblemRequest < Internal::Types::Model
      field :problem_name, -> { String }, optional: false, nullable: false, api_name: "problemName"
      field :problem_description, -> { Seed::Types::ProblemDescription }, optional: false, nullable: false, api_name: "problemDescription"
      field :files, -> { Internal::Types::Hash[String, Seed::Types::ProblemFiles] }, optional: false, nullable: false
      field :input_params, -> { Internal::Types::Array[Seed::Types::VariableTypeAndName] }, optional: false, nullable: false, api_name: "inputParams"
      field :output_type, -> { Seed::Types::VariableType }, optional: false, nullable: false, api_name: "outputType"
      field :testcases, -> { Internal::Types::Array[Seed::Types::TestCaseWithExpectedResult] }, optional: false, nullable: false
      field :method_name, -> { String }, optional: false, nullable: false, api_name: "methodName"
    end
  end
end
